pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -431,6 +437,70 @@ protected CompletableFuture<TransactionPendingAckInternalStats> internalGetPendi
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_SEGMENTS);
+                        return getTxnSnapshotInternalStats(segmentTopic, metadata)
+                                .thenApply(snapshotSystemTopicInternalStats -> {
+                                    transactionBufferInternalStats.segmentInternalStats =
+                                            snapshotSystemTopicInternalStats;
+                                    return transactionBufferInternalStats;
+                                }).thenCompose(ignore -> {
+                                    TopicName indexTopic = TopicName.get(TopicDomain.persistent.toString(),
+                                            namespaceName,
+                                            SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_INDEXES);
+                                    return getTxnSnapshotInternalStats(indexTopic, metadata)
+                                            .thenApply(indexStats -> {
+                                                transactionBufferInternalStats.segmentIndexInternalStats = indexStats;
+                                                return transactionBufferInternalStats;
+                                            });
+                                });
+                    } else if (snapshotType == TransactionBuffer.SnapshotType.Single) {
+                        transactionBufferInternalStats.snapshotType = snapshotType.toString();
+                        TopicName singleSnapshotTopic = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT);
+                        return getTxnSnapshotInternalStats(singleSnapshotTopic, metadata)
+                                .thenApply(snapshotSystemTopicInternalStats -> {
+                                   transactionBufferInternalStats.singleSnapshotSystemTopicInternalStats =
+                                           snapshotSystemTopicInternalStats;
+                                   return transactionBufferInternalStats;
+                                });
+                    }
+                    return FutureUtil.failedFuture(new RestException(INTERNAL_SERVER_ERROR, "Unknown SnapshotType "
+                            + snapshotType));
+                });
+    }
+
+    private CompletableFuture<SnapshotSystemTopicInternalStats> getTxnSnapshotInternalStats(TopicName topicName,
+                                                                                            boolean metadata) {
+        final PulsarAdmin admin;
+        try {
+            admin = pulsar().getAdminClient();
+        } catch (PulsarServerException e) {
+            return FutureUtil.failedFuture(new RestException(e));
+        }
+        NamespaceService ns = pulsar().getNamespaceService();
+        return ns.isServiceUnitOwnedAsync(topicName)
```

Comment:
Why need to check the ownership here? The result `isOwner` is not used, and the method `admin.topics().getInternalStatsAsync` can redirect requests by itself.